Sources/BraintreeCoreSwift/BTAnalyticsService.swift

```diff
@@ -0,0 +1,174 @@
+                    return
+                }
+
+                var willPostAnalyticsEvent: Bool = false
+
+                self.analyticsSessions.keys.forEach { sessionID in
+                    var session = self.analyticsSessions[sessionID]
+
+                    willPostAnalyticsEvent = true
+
+                    let metadataParameters: [String: Any] = [
+                        "sessionId": session?.sessionID ?? "",
+                        "integrationType": session?.integration ?? "",
+                        "source": session?.source ?? ""
+                    ]
+                        .merging(session?.metadataParameters ?? [:]) { $1 }
+
+                    var postParameters: [String: Any] = [:]
+
+                    if let sessionEvents = session?.events {
+                        // Map array of BTAnalyticsEvent to JSON
+                        postParameters["analytics"] = sessionEvents.map { $0.json }
+                    }
+
+                    postParameters["_meta"] = metadataParameters
+
+                    if let authorizationFingerprint = self.apiClient.clientToken?.authorizationFingerprint {
+                        postParameters["authorization_fingerprint"] = authorizationFingerprint
+                    } else if let tokenizationKey = self.apiClient.tokenizationKey {
+                        postParameters["tokenization_key"] = tokenizationKey
+                    }
+
+                    session?.events.removeAll()
+
+                    self.http?.post("/", parameters: postParameters) { body, response, error in
+                        if let error {
+                            completion(error)
+                        }
+                    }
+                }
+
+                if !willPostAnalyticsEvent {
+                    completion(nil)
+                }
+
+                completion(nil)
+            }
+        }
+    }
+
+    // MARK: - Helpers
+
+    func enqueueEvent(_ eventKind: String) {
+        let timestampInMilliseconds = Date().timeIntervalSince1970 * 1000
+        let event = BTAnalyticsEvent(kind: eventKind, timestamp: timestampInMilliseconds)
+        let session = BTAnalyticsSession(
+            sessionID: apiClient.metadata.sessionID,
+            source: apiClient.metadata.sourceString,
+            integration: apiClient.metadata.integrationString
+        )
+
+        sessionsQueue.async {
+            if self.analyticsSessions[session.sessionID] == nil {
+                self.analyticsSessions[session.sessionID] = session
+            }
+
+            self.analyticsSessions[session.sessionID]?.events.append(event)
+        }
+    }
+
+    func checkFlushThreshold() {
+        var eventCount = 0
+
+        sessionsQueue.sync {
+            analyticsSessions.values.forEach { analyticsSession in
+                eventCount += analyticsSession.events.count
+            }
+        }
+
+        if eventCount >= flushThreshold {
+            flush()
+        }
+    }
```

Comment:
This function name doesn't fully express that it calls `flush()`. What about `flushIfAtThreshold`?